libr/bin/format/elf/elf.c

```diff
@@ -2627,150 +2635,276 @@ char *Elf_(r_bin_elf_get_rpath)(ELFOBJ *bin) {
 	return ret;
 }
 
+static void fix_rva_and_offset(ELFOBJ *bin, RBinElfReloc *r, ut64 offset_table) {
+	if (is_bin_etrel (bin)) {
+		r->rva = offset_table + r->offset;
+		r->rva = Elf_(r_bin_elf_p2v) (bin, r->rva);
+	} else {
+		r->rva = r->offset;
+		r->offset = Elf_(r_bin_elf_v2p) (bin, r->offset);
+	}
+}
+
+static int read_reloc(ELFOBJ *bin, RBinElfReloc *r, int is_rela, ut64 offset) {
+	size_t size_struct;
+
+	if (is_rela) {
```

Comment:
actually, i'm not sure this works as expected, because `read_reloc` receives `DT_REL` and `DT_RELA` and not a boolean. I think you can probably do `size_t size_struct = get_size_rel_mode (is_rela)`.